<?xml version="1.0"?>
<launch>
  <!-- Load waypoint parameters -->
  <arg name="waypoints_file" default="$(find continuous_navigation)/config/waypoints.yaml" />
  <arg name="global_frame" default="map" />
  <arg name="cycle_waypoints" default="true" />
  <arg name="pause_between_waypoints" default="false" />
  <arg name="pause_duration" default="2.0" />
  <arg name="goal_timeout" default="60.0" />
  <arg name="num_manual_goals" default="2" />
  <arg name="auto_start" default="true" />
  <arg name="manual_waypoints_file" default="$(find continuous_navigation)/config/manual_waypoints.yaml" />
  <arg name="save_manual_waypoints" default="true" />
  
  <!-- Launch the waypoint navigator node -->
  <node pkg="continuous_navigation" type="waypoint_navigator" name="waypoint_navigator" output="screen">
    <param name="waypoints_file" value="$(arg waypoints_file)" />
    <param name="global_frame" value="$(arg global_frame)" />
    <param name="cycle_waypoints" value="$(arg cycle_waypoints)" />
    <param name="pause_between_waypoints" value="$(arg pause_between_waypoints)" />
    <param name="pause_duration" value="$(arg pause_duration)" />
    <param name="goal_timeout" value="$(arg goal_timeout)" />
    <param name="num_manual_goals" value="$(arg num_manual_goals)" />
    <param name="auto_start" value="$(arg auto_start)" />
    <param name="manual_waypoints_file" value="$(arg manual_waypoints_file)" />
    <param name="save_manual_waypoints" value="$(arg save_manual_waypoints)" />
  </node>
</launch> 